src/server.rs

```diff
@@ -0,0 +1,46 @@
+//! The application's web server.
+//! Serves both api and web traffic.
+
+use crate::{
+    api::{self, ApiState, ApiStateInner},
+    models::forum::Forum,
+    resources::DatabaseClient,
+    web::{pages, WebState, WebStateInner},
+};
+use axum::{Extension, Router};
+
+pub async fn run() -> color_eyre::Result<Router> {
+    let database = DatabaseClient::new()?;
+    let database_health_check =
+        health::PeriodicChecker::new(database.clone(), health::Config::default());
+    tokio::spawn(database_health_check.clone().run());
```

Comment:
**question**: There are a lot of clones here. The way I would think this will be implemented later is with a database connection pool; in that case we would just be giving this a reference to the pool, would we not? Ownership over the connection certainly seems overkill to me; at most I would expect it to have a mutable reference, perhaps to an `Arc<Mutex<DatabasePool>>`  or something similar.

Reply:
Made #4 to track this, as I don't expect this to be fleshed out in this PR.

Reply:
thanks for doing that!